strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/repository/remote/heartbeat/RemoteRepositoryAlivenessCacheManagerTest.java

```diff
@@ -0,0 +1,157 @@
+import org.carlspring.strongbox.testing.NullLayoutConfiguration;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.context.annotation.*;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestExecutionListeners;
+
+import javax.inject.Inject;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+@ActiveProfiles({ "test",
+                  "RemoteRepositoryAlivenessCacheManagerTestConfig" })
+@TestExecutionListeners(listeners = { CacheManagerTestExecutionListener.class },
+                        mergeMode = TestExecutionListeners.MergeMode.MERGE_WITH_DEFAULTS)
+public class RemoteRepositoryAlivenessCacheManagerTest
+{
+
+    @Inject
+    private CacheManager cacheManager;
+
+    private RemoteRepositoryAlivenessCacheManager remoteRepositoryAlivenessCacheManager;
+
+    private final static String REMOTE_REPOSITORY_URL = "http://mocked-remote-repository-url";
+
+    @BeforeEach
+    void setUp()
+    {
+        remoteRepositoryAlivenessCacheManager = new RemoteRepositoryAlivenessCacheManager(cacheManager);
+    }
+
+    @Test
+    public void isAliveShouldReturnTrueWhenRemoteRepositoryNotCached()
+    {
+        RemoteRepository remoteRepository = createMockedRemoteRepository();
+
+        boolean alive = remoteRepositoryAlivenessCacheManager.isAlive(remoteRepository);
+
+        assertTrue(alive);
+    }
+
+    @Test
+    public void isAliveShouldReturnTrueWhenRemoteRepositoryCachedValueIsTrue()
+    {
+        RemoteRepository remoteRepository = createMockedRemoteRepository();
+
+        initializeCache(true);
+
+        boolean alive = remoteRepositoryAlivenessCacheManager.isAlive(remoteRepository);
+
+        assertTrue(alive);
+    }
+
+    @Test
+    public void isAliveShouldReturnFalseWhenRemoteRepositoryCachedValueIsFalse()
+    {
+        RemoteRepository remoteRepository = createMockedRemoteRepository();
+
+        initializeCache(false);
+        remoteRepositoryAlivenessCacheManager.put(remoteRepository, false);
+
+        boolean alive = remoteRepositoryAlivenessCacheManager.isAlive(remoteRepository);
+
+        assertFalse(alive);
+    }
+
+    @Test
+    public void isAliveShouldReturnTrueWhenRemoteRepositoryCachedValueIsExpired()
+    {
+        // TODO: Figure out how to forcibly expire a cached item.
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideCacheValues")
+    public void putShouldChangeCachedValueImmediately(Boolean initialCacheValue,
```

Comment:
this parameterized method looks very nice :+1:

Reply:
Thank you :-)